src/clr-angular/forms/common/form.ts

```diff
@@ -28,4 +52,74 @@ export class ClrForm {
   markAsTouched() {
     this.markControlService.markAsTouched();
   }
+
+  @ContentChildren(ClrLabel, { descendants: true })
+  labels: QueryList<ClrLabel>;
+
+  @HostListener('submit')
+  onFormSubmit() {
+    this.markAsTouched();
+    if (isPlatformBrowser(this.platformId)) {
+      this.invalidControls = Array.from(this.el.nativeElement.querySelectorAll('.ng-invalid'));
+      if (this.invalidControls.length > 0) {
+        this.invalidControls[0].focus();
+        this.updateAriaLive();
+      } else {
+        this.resetAriaLive();
+      }
+    }
+  }
+
+  ngOnInit() {
+    this.createAriaLiveElement();
+  }
+
+  private resetAriaLive() {
+    this.ariaLiveElement.innerHTML = '';
+  }
+
+  private updateAriaLive(): void {
+    if (this.invalidControls.length === 0) {
+      return;
+    }
+
+    this.ariaLiveText = this.commonStrings.keys.formErrorSummary;
+    this.resetAriaLive();
+
+    this.ariaLiveElement.textContent =
+      this.commonStrings.keys.formErrorSummary +
+      // Only use labels that have a 'for' attribute
+      this.labels
+        .filter(label => label.forAttr !== undefined)
+        // Only use labels that have a control id that matches the for attribute
+        .filter(label => this.invalidControls.filter(control => control.id === label.forAttr))
+        // Grab the label text for the label with an associated error
+        .map(label => label.labelText)
+        // join all of the label text with associated errors together and seperate them with spaces
+        .join(' ');
+  }
+
+  private createAriaLiveElement() {
+    if (isPlatformBrowser(this.platformId)) {
+      // ClrForms need to have aria live element for screen readers.
+      this.ariaLiveElement = document.createElement('div');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-live', 'polite');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-live', 'polite');
+      this.renderer.setAttribute(this.ariaLiveElement, 'role', 'status');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-atomic', 'true');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-relevent', 'text');
+      this.renderer.addClass(this.ariaLiveElement, 'clr-sr-only');
+      this.renderer.insertBefore(this.el.nativeElement, this.ariaLiveElement, this.el.nativeElement.firstChild);
+    }
+  }
+
+  private removeAriaLiveElement() {
+    if (isPlatformBrowser(this.platformId)) {
+      this.renderer.removeChild(this.el.nativeElement, this.ariaLiveElement);
```

Comment:
For creating and removing an element, I've learned that it's better to use Native DOM API. It's due to the weird DOM leakage issue in Angular I filed a while ago: https://github.com/angular/angular/issues/26954. So we started using Native `.removeChild` in FocusTrapDirective. In your case, it would be just:

```
this.el.nativeElement.removeChild(this.ariaLiveElement)
```

Also, just in case, we need to wipe out the property value we used for the element: `delete this.ariaLiveElement`;

Reply:
If I read the issue correctly it is only an issue when testing. The memory leak will not occur in production or dev serve environment. Right?  

Anyway, I'll go back to removing it with the native DOM method like I had to begin with and add the delete.

Reply:
Yeah, it's only testing related issue. You can still use the attribute/class manipulation Renderer2 APIs, but only creation/deletion of elements need to be implemented with Native DOM APIs.